src/Common/src/TypeSystem/Common/TypeSystemContext.cs

```diff
@@ -687,5 +687,13 @@ public virtual VirtualMethodEnumerationAlgorithm GetVirtualMethodEnumerationAlgo
             // Type system contexts that support this need to override this.
             throw new NotSupportedException();
         }
+
+        // Abstraction to allow different runtimes to have different policy about which fields are 
+        // in the GC static region, and which are not
+        public virtual bool ComputeHasGCStaticsBase(FieldDesc field)
```

Comment:
Make this `protected internal`? Users outside of the type system assembly shouldn't care about this or see this in IntelliSense.